api/src/main/java/org/openmrs/module/fhir/api/client/FHIRClient.java

```diff
@@ -23,32 +23,24 @@
     private static final String ACCEPT_MIME_TYPE = "application/json";
 
     private RestTemplate restTemplate = new RestTemplate();
-    IParser parser = FhirContext.forDstu3().newJsonParser();
 
     public FHIRClient(ClientHttpRequestFactory clientHttpRequestFactory) {
         restTemplate.setRequestFactory(clientHttpRequestFactory);
     }
 
     @Override
-    public Object getObject(String category, String url, String username, String password) throws HttpClientErrorException {
+    public Object getObject(String category, String url, String username, String password)
+            throws RestClientException {
         prepareRestTemplate(username, password);
-        String stringObject = "";
-        try {
-            stringObject = restTemplate.getForObject(url, resolveCategory(category)).toString();
-        } catch(HttpClientErrorException e) {
-            log.error(String.format("Resource %s not found", category));
-        }
-        return convertStringToFHIRObject(resolveCategory(category), stringObject);
+        return restTemplate.getForObject(url, resolveCategory(category));
     }
 
-    private Object convertStringToFHIRObject(Class classType, String stringObject) {
-        Object result = "";
-        try {
-            result = parser.parseResource(classType, stringObject);
-        } catch(DataFormatException e) {
-            log.error(String.format("Could not parse String to Object: %s", stringObject));
-        }
-        return result;
+    @Override
+    public ResponseEntity<String> postObject(String category, String url, String username, String password, Object object)
+            throws RestClientException {
+        prepareRestTemplate(username, password);
+        IBaseResource baseResource = (IBaseResource) object;
+        return restTemplate.postForEntity(category, baseResource, String.class);
```

Comment:
I think that first parameter should be URL not category.

Reply:
Thanks a lot! Thats my mislook